src/WebJobs.Script.WebHost/Configuration/ScriptApplicationHostOptionsSetup.cs

```diff
@@ -65,14 +68,91 @@ public void Configure(string name, ScriptApplicationHostOptions options)
                 options.IsStandbyConfiguration = true;
             }
 
-            options.IsFileSystemReadOnly = IsZipDeployment();
+            options.IsFileSystemReadOnly = IsZipDeployment(options);
         }
 
-        private bool IsZipDeployment()
+        private bool IsZipDeployment(ScriptApplicationHostOptions options)
```

Comment:
>IsZipDeployment(ScriptApplicationHostOptions options) [](http://example.com/codeflow?start=21&length=53)

Make this
IsZipDeployment(out bool isScmRunFromPackage)

to make sure we are explicitly setting a value in all code paths